src/video_core/command_classes/vic.cpp

```diff
@@ -0,0 +1,189 @@
+
+        if (scaler_ctx == nullptr || frame->width != scaler_width ||
+            frame->height != scaler_height) {
+            sws_freeContext(scaler_ctx);
+            scaler_ctx = nullptr;
+            // FFmpeg returns all frames in YUV420, convert it into RGBA format
+            scaler_ctx =
+                sws_getContext(frame->width, frame->height, AV_PIX_FMT_YUV420P, frame->width,
+                               frame->height, AV_PIX_FMT_RGBA, 0, nullptr, nullptr, nullptr);
+
+            scaler_width = frame->width;
+            scaler_height = frame->height;
+        }
+
+        // Get RGB frame
+        const std::size_t linear_size = frame->width * frame->height * 4;
+        u8* rgb_frame_buffer = static_cast<u8*>(av_malloc(linear_size));
+        const std::array<int, 1> rgb_stride{frame->width * 4};
+        const std::array<u8*, 1> rgb_frame_buf_addr{rgb_frame_buffer};
+
+        sws_scale(scaler_ctx, frame->data, frame->linesize, 0, frame->height,
+                  rgb_frame_buf_addr.data(), rgb_stride.data());
+
+        const u32 blk_kind = static_cast<u32>(config.block_linear_kind);
+        if (blk_kind != 0) {
+            // swizzle pitch linear to block linear
+            const u32 block_height = static_cast<u32>(config.block_linear_height_log2);
+            const auto size = Tegra::Texture::CalculateSize(true, 4, frame->width, frame->height, 1,
+                                                            block_height, 0);
+            std::vector<u8> swizzled_data(size);
+            Tegra::Texture::CopySwizzledData(frame->width, frame->height, 1, 4, 4,
+                                             swizzled_data.data(), rgb_frame_buffer, false,
+                                             block_height, 0, 1);
+
+            gpu.Maxwell3D().OnMemoryWrite();
+            gpu.MemoryManager().WriteBlock(output_surface_luma_address, swizzled_data.data(), size);
+        } else {
+            // send pitch linear frame
+            gpu.Maxwell3D().OnMemoryWrite();
+            gpu.MemoryManager().WriteBlock(output_surface_luma_address, rgb_frame_buf_addr.data(),
+                                           linear_size);
+        }
+
+        av_free(rgb_frame_buffer);
+        break;
+    }
+    case VideoPixelFormat::Yuv420: {
+        LOG_DEBUG(Service_NVDRV, "Writing YUV420 Frame");
+
+        const auto frame = nvdec_processor->GetFrame();
+
+        if (!frame || frame->width == 0 || frame->height == 0) {
+            return;
+        }
+
+        const std::size_t surface_width = config.surface_width_minus1 + 1;
+        const std::size_t surface_height = config.surface_height_minus1 + 1;
+        const std::size_t src_half_width = frame->width / 2;
+        const std::size_t half_width = surface_width / 2;
+        const std::size_t half_height = config.surface_height_minus1 / 2;
+        const std::size_t aligned_width = (surface_width + 0xff) & ~0xff;
+
+        const auto luma_ptr = frame->data[0];
```

Comment:
```suggestion
        const auto* luma_ptr = frame->data[0];
```

Ditto for the others if they happen to be pointers as well.